components/new_channel_modal/new_channel_modal.jsx

```diff
@@ -264,7 +260,7 @@ export default class NewChannelModal extends React.PureComponent {
                                 <div className='col-sm-9'>
                                     <input
                                         id={inputPrefixId + 'Name'}
-                                        onChange={this.handleChange}
+                                        onChange={() => this.handleChange()}
```

Comment:
Should not pass anonymous function as props. This should be like:
```
onChange={this.handleChange}
```

then `handleChange` should be:
```
handleChange = () => {
    ...
}
```

Please do the same pattern to `onEnterKeyDown` and `handleSubmit`.

Reply:
Oh I didn't know you could do that. Cool, I learned something today!